vespajlib/src/main/java/com/yahoo/collections/MethodCache.java

```diff
@@ -35,7 +35,7 @@ public Method get(Object object) {
     }
     public Method get(Object object, Consumer<String> onPut) {
         Method m = cache.get(object.getClass().getName());
-        if ( ! m.getDeclaringClass().isAssignableFrom(object.getClass())) {
+        if ( m != null && ! m.getDeclaringClass().isAssignableFrom(object.getClass())) {
```

Comment:
If these calls are slow, we could also just use a composite value of (class, method), and compare class for identity.